tests/integration/deprecated_syscalls_tests.rs

```diff
@@ -491,3 +491,45 @@ async fn test_syscall_get_contract_address_cairo0(
     // We can only verify that the syscall is implemented and goes through.
     check_os_output_read_only_syscall(os_output, block_context);
 }
+
+#[rstest]
+// We need to use the multi_thread runtime to use task::block_in_place for sync -> async calls.
+#[tokio::test(flavor = "multi_thread", worker_threads = 1)]
+async fn test_syscall_send_message_to_l1_cairo0(
+    #[future] initial_state_cairo0: StarknetTestState,
+    block_context: BlockContext,
+    max_fee: Fee,
+) {
+    let initial_state = initial_state_cairo0.await;
+
+    let sender_address = initial_state.cairo0_contracts.get("account_with_dummy_validate").unwrap().address;
+    let test_contract = initial_state.cairo0_contracts.get("test_contract").unwrap();
+    let contract_address = test_contract.address;
+
+    let tx_version = TransactionVersion::ZERO;
+
+    let to_address = stark_felt!(1234_u16);
+
+    let entrypoint_args = &[vec![to_address]].concat();
+
+    let mut nonce_manager = NonceManager::default();
+    let tx = test_utils::account_invoke_tx(invoke_tx_args! {
+        max_fee,
+        sender_address: sender_address,
+        calldata: create_calldata(contract_address, "send_message", entrypoint_args),
+        version: tx_version,
+        nonce: nonce_manager.next(sender_address),
+    });
+
+    let txs = vec![tx];
+
+    let (_pie, _os_output) = execute_txs_and_run_os(
```

Comment:
Okay so here it would be nice to check if the L1 message appears in the OS output. `StarknetOsOuput` has a `messages_to_l1` field, I would expect that the message appears there. Could you check?